drivers/cc110x/cc110x_netdev.c

```diff
@@ -0,0 +1,610 @@
+        return -1;
+    }
+
+    cc110x_write(dev, CC110X_REG_IOCFG0, CC110X_GDO_CONSTANT_LOW);
+    if (gpio_read(dev->params.gdo0)) {
+        DEBUG("[cc110x] GDO0 does not respond (check wiring!)\n");
+        return -1;
+    }
+
+    /* Restore default GDO2 & GDO0 config */
+    cc110x_write(dev, CC110X_REG_IOCFG2, cc110x_conf[CC110X_REG_IOCFG2]);
+    cc110x_write(dev, CC110X_REG_IOCFG0, cc110x_conf[CC110X_REG_IOCFG0]);
+
+    return 0;
+}
+
+static int cc110x_init(netdev_t *netdev)
+{
+    cc110x_t *dev = (cc110x_t *)netdev;
+    /* Use locked mutex to block thread on TX and un-block from ISR */
+    mutex_init(&dev->isr_signal);
+    mutex_lock(&dev->isr_signal);
+
+    /* Make sure the crystal is stable and the chip ready */
+    if (cc110x_power_on(dev)) {
+        DEBUG("[cc110x] netdev_driver_t::init(): Failed to pull CS pin low\n");
+        return -EIO;
+    }
+
+    if (cc110x_acquire(dev) != SPI_OK) {
+        DEBUG("[cc110x] netdev_driver_t::init(): Failed to setup/acquire SPI "
+              "interface\n");
+        return -EIO;
+    }
+
+    /* Performing a reset of the transceiver to get it in a known state */
+    cc110x_cmd(dev, CC110X_STROBE_RESET);
+    cc110x_release(dev);
+
+    /* Again, make sure the crystal is stable and the chip ready */
+    if (cc110x_power_on(dev)) {
+        DEBUG("[cc110x] netdev_driver_t::init(): Failed to pull CS pin low "
+              "after reset\n");
+        return -EIO;
+    }
+
+    if (cc110x_acquire(dev) != SPI_OK) {
+        DEBUG("[cc110x] netdev_driver_t::init(): Failed to setup/acquire SPI "
+              "interface after reset\n");
+        return -EIO;
+    }
+
+    if (check_device(dev)) {
+        DEBUG("[cc110x] netdev_driver_t::init(): Device check failed\n");
+        cc110x_release(dev);
+        return -ENOTSUP;
+    }
+
+    /* Upload the main configuration */
+    cc110x_burst_write(dev, CC110X_CONF_START, cc110x_conf, CC110X_CONF_SIZE);
+    /* Set TX power to match uploaded configuration */
+    dev->tx_power = CC110X_TX_POWER_0_DBM;
+
+    /* Upload the poorly documented magic numbers obtained via SmartRF Studio */
+    cc110x_burst_write(dev, CC110X_REG_TEST2, cc110x_magic_registers,
+                       sizeof(cc110x_magic_registers));
+
+    /* Setup the selected PA_TABLE */
+    cc110x_burst_write(dev, CC110X_MULTIREG_PATABLE,
+                       dev->params.patable->data, CC110X_PATABLE_LEN);
+
+    /* Verify main config, magic numbers and PA_TABLE correctly uploaded */
+    if (check_config(dev)) {
+        cc110x_release(dev);
+        return -EIO;
+    }
+
+    /* Verify that pins GDO2 and GDO0 are correctly connected */
+    if (check_gdo_pins(dev)) {
+        cc110x_release(dev);
+        return -EIO;
+    }
+
+    /* Setup the layer 2 address, but do not accept CC110X_L2ADDR_AUTO (which
+     * has the value 0x00 and is used for broadcast)
+     */
+    dev->addr = dev->params.l2addr;
+    while (dev->addr == CC110X_L2ADDR_AUTO) {
```

Comment:
The L2 address can be specified in the `cc110x_params_t`. The `CC110X_L2ADDR_AUTO` has the value of the broadcast address contains, as no transceiver should ever use this as device address.

`luid_get()` can basically be seen as a pseudo RNG seeded with the CPU ID (so that the same board will get the same L2 address on every boot). That RNG is free to return the broadcast address by chance. So in that case, we just roll the dice again until finally a valid address comes out (so every value but the broadcast address).